### PR TITLE
Diagnostic Aggregator Testing

### PR DESCRIPTION
**Pull request by norro**
This PR introduces a demo and two tests, but...:
One of the tests is supposed to test the analyzer creation of the aggregator from a yaml file. This one, however, is failing since the output it is supposed to match somehow only appears _after_ the test times out.
The second test is succeeding, but only checks for output of the bond state machine, which somehow already appears _before_ the test times out, but is not meaningful to test.

@Karsten1987: This kind-of fixes [this](https://github.com/ros/diagnostics/pull/118#issuecomment-519280962) as is shows how the tests should look like IMHO. Do you have an idea why the logging output is stalled until launch-testing sends a `SIGINT` to the process under test (`aggregator_node`)?
The stalling does not happen, if I start the `aggregator_node` with `ros2 run` (see the [demo](https://github.com/boschresearch/ros2_diagnostics/blob/ros2_migrate_diagnostic_aggregator_testing/diagnostic_aggregator/demo/demo.md)), so it might be me using ros2 launch-testing not in the intended way.

### CODE REVIEW

**Comment by norro** [comment]
Four tests working, testing creation of various analyzers and analyzer groups from yaml specification.
@Karsten1987: This should fix https://github.com/ros/diagnostics/pull/118

File: diagnostic_aggregator/src/aggregator.cpp [hunk 1]
```diff
@@ -93,6 +93,9 @@ Aggregator::Aggregator()
   // Last analyzer handles remaining data
   other_analyzer_ = std::make_unique<OtherAnalyzer>(other_as_errors);
   other_analyzer_->init(base_path_);  // This always returns true
+  
+  // @todo(anordman): This cout somehow is necessary to continue screen output for testing
+  std::cout << "Aggregator created, starting analysis..." << std::endl;
```
Comment (Karsten1987) [inline diagnostic_aggregator/src/aggregator.cpp]: I can reproduce this. Do you have an idea why this print changes things?

**Review by Karsten1987 — COMMENTED** [review]
Generally looks good to me. I'd like to have the `std::cout` though addressed. 

also I went ahead and pushed 087d391 to address some failing linters.

File: diagnostic_aggregator/src/aggregator.cpp (hunk 1 above)
Comment (norro) [reply diagnostic_aggregator/src/aggregator.cpp]: The general problem seems to be that stdout is not flushed when only log statements get written. During debugging I found a lot of threads addressing this general topic, e.g., raising log level in launch, forcing flush of stdout within launch, etc. See https://github.com/ros2/rcutils/issues/70 and https://github.com/ros2/rcutils/pull/98 for example.
None of the suggested work-arounds I found worked for me, though, so I should probably report another issue. I am unsure where it should go, though. To rcutils or maybe to ros2/launch?

Comment (norro) [reply diagnostic_aggregator/src/aggregator.cpp]: I was finally able to solve this problem with ``ExecuteProcess``, which can be told to emulate a tty (``emulate_tty=True``, which is ``False`` by default).
[test/create_analyzers.launch.py.in#L30](https://github.com/boschresearch/ros2_diagnostics/blob/ros2_migrate_diagnostic_aggregator_testing/diagnostic_aggregator/test/create_analyzers.launch.py.in#L30)

**Comment by Karsten1987** [comment]
@norro I've seen you've updated your branch here. Is this good to be merged in your original PR?
I am about to cut a release for eloquent and was wondering whether that feature should be part of it.

**Comment by norro** [comment]
@Karsten1987 yes, this could/should be merged into the original PR, as it now has [a small demo](https://github.com/boschresearch/ros2_diagnostics/blob/ros2_migrate_diagnostic_aggregator_testing/diagnostic_aggregator/demo/demo.md) and working tests.

**Pull request merged**